packages/shell-api/src/collection.ts

```diff
@@ -2033,4 +2034,29 @@ export default class Collection extends ShellApiWithMongoClass {
     this._emitCollectionApiCall('unhideIndex');
     return setHideIndex(this, index, false);
   }
+
+  @returnsPromise
+  @topologies([Topologies.ReplSet, Topologies.Sharded])
+  @apiVersions([])
+  async analyzeShardKey(key: Document): Promise<Document> {
+    assertArgsDefinedType([key], [true], 'Collection.analyzeShardKey');
+    this._emitCollectionApiCall('analyzeShardKey', { key });
+    return await this._database._runAdminCommand({
+      analyzeShardKey: this.getFullName(),
+      key
+    });
+  }
+
+  @returnsPromise
+  @topologies([Topologies.ReplSet, Topologies.Sharded])
+  @apiVersions([])
+  async configureQueryAnalyzer({ mode, sampleRate }: { mode: 'full' | 'off', sampleRate?: number }): Promise<Document> {
```

Comment:
sampleRate is only for mode: 'full' in which case it is required. Not sure how best to represent that in typescript without excessive branching below.

Reply:
I think that would be 

```suggestion
  async configureQueryAnalyzer({ mode, sampleRate }: { mode: 'full', sampleRate: number } | { mode: 'off', sampleRate?: never }): Promise<Document> {
```

but I really think we don’t need to worry about those details here.